aws-s3-transfer-manager/src/operation/upload/service.rs

```diff
@@ -96,67 +97,63 @@ pub(super) fn upload_part_service(
 /// * stream - the body input stream
 /// * part_size - the part_size for each part
 pub(super) fn distribute_work(
-    handle: &mut UploadHandle,
+    mpu_data: &mut MultipartUploadData,
+    ctx: UploadContext,
     stream: InputStream,
     part_size: u64,
+    upload_id: String,
```

Comment:
isn't upload id part of `mpu_data`? 

Also the whole point of `UploadContext` was so to thread through all the state needed for an operation...

Reply:
> isn't upload id part of mpu_data?

Thanks, fixed by not passing upload_id explicitly. 

> Also the whole point of UploadContext was so to thread through all the state needed for an operation...

Yeah, but the problem was that we no longer have mutable access to UploadContext in the createMPU call, because we do not own the handle at that point.